docs/development/howto.rst

```diff
@@ -907,6 +907,29 @@ putting this in ``docs/conf.py`` can also help sometimes::
     warnings.showwarning = warn_with_traceback
 
 
+Providing summary info for class instances
+------------------------------------------
+
+If you want to add a method to provide some basic information about a class instance,
+you should use the Python ``__str__`` method.
+
+.. code-block:: python
+
+    class foo(object):
+        def __init__(self, a):
+            self.a = a
+
+        def __str__(self):
+            ss = 'Summary Info about class foo\n'
+            ss += '{:.2f}'.format(self.a)
+
+If you want to add configurable info output, please provide a method ``summary``.
+In this case the ``__str__`` method should be a call to ``summary`` with default
+parameters. Do not use an ``info`` method, since this would lead to conflicts
+for some classes in Gammapy (e.g. classes that inherit the ``info`` method from
+``astropy.table.Table``.
```

Comment:
I think the main case where we have a lot and complex text representation is:
https://gammapy.readthedocs.org/en/latest/api/gammapy.catalog.SourceCatalogObjectHGPS.html

I think `SourceCatalogObjectHGPS.print_info` should become `SourceCatalogObjectHGPS.summary` with a multiple-choice option `info = {'basic', 'associations', ...}` and the other methods become private (by adding an underscore)?

An example is always better than a text description, so in this case I'd suggest to link to `SourceCatalogObjectHGPS.summary` from here.